lib/plugins/aws/package/compile/events/apiGateway/lib/validate.js

```diff
@@ -470,4 +472,15 @@ module.exports = {
     const splitLambdaName = splitArn[splitArn.length - 1].split('-');
     return splitLambdaName[splitLambdaName.length - 1];
   },
+
+  validateRestApiRootResourceId() {
+    const apiGateway = this.serverless.service.provider.apiGateway;
+
+    if (apiGateway && apiGateway.restApiId && !apiGateway.restApiRootResourceId) {
+      throw new ServerlessError(
+        "Configuration warning at 'provider.apiGateway': should have required property 'restApiRootResourceId'",
```

Comment:
We're throwing an _error_, so it's not a warning (which is permissive by nature).

Let's reword it to:

```
Missing required "provider.apiGateway.restApiRootResourceId" property (needed if "provider.apiGateway.restApiId" is provided)
```